### DISCUSSION TITLE
Can someone help me with an example of a React-Native app with V2 implementation that connects to metamask?

### DISCUSSION THREAD

**Discussion by AnanthanQB** [WalletConnect/walletconnect-monorepo#1818] [Developer Support]
Hi, I am working on a project that just needs to retrieve the user's public address from the metamask wallet into my project app. I got the app working with the V1 protocol but I couldn't get any viable guides for the WalletConnect V2. Can someone help me with this issue?

**Comment by 95decode** [comment]
Metamask plan to support Wallet Connect v2 by the end of Q1 2023.

V2 not yet supported

see [issue](https://github.com/MetaMask/metamask-mobile/issues/3957)

**Reply by crypblizz8** [reply]
Here is also some starter code for SignV2 for [react-native](https://github.com/WalletConnect/react-native-examples).

We will be improving this with more documentation + codebase in the coming weeks.
